torsiondrive/qm_engine.py

```diff
@@ -327,6 +330,114 @@ def load_native_output(self, filename='output.dat'):
         m.build_topology()
         return m
 
+class EnginexTB(QMEngine):
+    def load_input(self, input_file):
+        """ Load a xTB input file as a Molecule object into self.M
+        Only .xyz file format is supported for now.
+        The comment line is used as the input argument for xTB
+        Exmaple input file:
+
+        4
+        xTB arguments: --opt --chrg 0 --uhf 0 --gfn 2 --parallel 1
+        H  -0.90095  -0.50851  -0.76734
+        O  -0.72805   0.02496   0.02398
+        O   0.72762   0.03316  -0.02696
+        H   0.90782  -0.41394   0.81465
+        """
+        coords = []
+        elems = []
+        with open(input_file, 'r') as xTBin:
+            n_atoms = int(xTBin.readline())
+            comment = xTBin.readline().strip()
+            for line in range(n_atoms):
+                elem, x, y, z = xTBin.readline().split()
+                elems.append(elem)
+                coords.append((x, y, z))
+
+        if comment[:len('xTB arguments:')] == 'xTB arguments:':
+            cmd = comment[len('xTB arguments:'):]
+        else:
+            cmd = '--opt'
+            print("'xTB arguments:' not found in the command line, no extra arguments passed to xTB.")
+
+        if self.native_opt:
+            self.temp_type = "optimize"
+            assert '--opt' in cmd, "comment line should contain --opt command to use native opt"
+
+        if '--parallel' in cmd:
+            self.core = int(cmd[cmd.index('--parallel')+len('--parallel'):].split()[0])
+            pass
+        elif '-P' in cmd:
+            self.core = int(cmd[cmd.index('-P')+len('-P'):].split()[0])
+        else:
+            # Run xTB in single core
+            self.core = 1
+            cmd += ' --parallel 1 '
+
+        self.xTB_args = cmd
+        # here self.M can be and will be overwritten by external functions
+        self.M = Molecule()
+        self.M.elem = elems
+        self.M.xyzs = [np.array(coords, dtype=float)]
+        self.M.build_topology()
+
+    def write_input(self, filename='coords'):
+        """ Write output based on self.M, using only geometry of the first frame """
+        # Write the coordinate file
+        with open(filename + '.xyz', 'w') as outfile:
+            outfile.write('{}\n\n'.format(len(self.M.elem)))
+            for e, c in zip(self.M.elem, self.M.xyzs[0]):
+                outfile.write("{:<7} {:<13.7f} {:<13.7f} {:<13.7f}\n".format(e, c[0], c[1], c[2]))
+        with open(filename + '.inp', 'w') as outfile:
+            outfile.write(self.opt_restrain)
+
+    def optimize_native(self):
+        """ run the constrained optimization using native opt, in 2 steps:
+        1. write a optimization job input file.
+        2. run the job
+        """
+        if self.extra_constraints is not None:
+            raise RuntimeError('extra constraints not supported in xTB native optimizations')
+        # add the optking command
```

Comment:
Don't think that has to do anything with `xtb`.

Reply:
Added TorsionDrive xTB native optimizations to show that this is not a xTB issue but rather I don't have time to work on it.